src/core/utils/index.js

```diff
@@ -304,3 +304,13 @@ export const normalizeFileNameId = (filename) => {
 export const getDisplayName = (component) => {
   return component.displayName || component.name || 'Component';
 };
+
+export const getPreviewImage = (
+  previews,
+  { version = 0, full = true } = {},
+) => {
+  const previewsType = full ? 'image_url' : 'thumbnail_url';
+  return previews && previews.length > 0 && previews[version][previewsType]
+    ? previews[version][previewsType]
+    : null;
+};
```

Comment:
I've only used the function here in this related section.. but I'll set up a follow issue/PR to use this in the other places too

Reply:
Would it make sense to pass the `addon` object instead?

Can you find a way to reuse this helper function somewhere else?

Reply:
hmm..I am not sure I am understanding why we should pass in `addon` instead... But let me take another look...

Reply:
I think I gave you a bad hint about this function. Do you think we can reuse it somewhere else? The code looks similar but not exactly the same (there are calls to `isAllowedOrigin()` for instance).

Reply:
oh sorry I missed your comment above about isAllowedOrigin and I guess the hint too:) ...

Yeh I guess I missed that piece of it. I am honestly not sure why it's in some places but not others (isAllowedOrigin). I guess I can add it to the getPreviewImage function. Would that work?

Reply:
So I added this isAllowedOrigin to the function. It just adds an extra step which I thought would be okay.  I *think* now it should cover the other cases too. Let me know what you think.